### ISSUE TITLE
Custom type conversion broken when using with GenerateDataTypes

### ISSUE THREAD

**Issue by liiws** [linq2db/linq2db#3060]
Custom type conversion throws exception when generate model with "GenerateDataTypes = true;"

When saving to database then conversion from custom type to database type is not called.

This behavior  started in 3.0.0 linq2db version and still presents in 3.4.0. In 2.9.8 it works ok.

```
Exception message:
"Unable to cast object of type 'Linq2DbConvertionError.Blob16AsGuidType' to type 'System.Byte[]'."

Stack trace:
System.InvalidCastException: Unable to cast object of type 'Linq2DbConvertionError.Blob16AsGuidType' to type 'System.Byte[]'.
   at System.Data.SQLite.SQLiteStatement.BindParameter(Int32 index, SQLiteParameter param)
   at System.Data.SQLite.SQLiteStatement.BindParameters()
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()
   at System.Data.SQLite.SQLiteCommand.GetStatement(Int32 index)
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery()
   at LinqToDB.Data.DataConnection.ExecuteNonQuery(IDbCommand command)
   at LinqToDB.Data.DataConnection.ExecuteNonQuery()
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteNonQueryImpl(DataConnection dataConnection, ExecutionPreparedQuery executionQuery)
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteNonQuery()
   at LinqToDB.Linq.QueryRunner.NonQueryQuery(Query query, IDataContext dataContext, Expression expr, Object[] parameters, Object[] preambles)
   at LinqToDB.Linq.QueryRunner.<>c__DisplayClass31_0.<SetNonQueryQuery>b__0(IDataContext db, Expression expr, Object[] ps, Object[] preambles)
   at LinqToDB.Linq.QueryRunner.Update`1.Query(IDataContext dataContext, T obj, UpdateColumnFilter`1 columnFilter, String tableName, String serverName, String databaseName, String schemaName, TableOptions tableOptions)
   at LinqToDB.DataExtensions.Update[T](IDataContext dataContext, T obj, UpdateColumnFilter`1 columnFilter, String tableName, String databaseName, String schemaName, String serverName, TableOptions tableOptions)
   at LinqToDB.DataExtensions.Update[T](IDataContext dataContext, T obj, String tableName, String databaseName, String schemaName, String serverName, TableOptions tableOptions)
   at Linq2DbConvertionError.Program.Main(String[] args) in D:\DOCUMENTS\PROGRMG\dotnet\TEST\Linq2DbConvertionError\Linq2DbConvertionError\Program.cs:line 42
```

### Steps to reproduce

Solution is attached.
1) Open it. Copy "example.db3" to "C:\tmp". If path is different then change it in Program.cs
2) In MyDb.tt comment "GenerateDataTypes = true;" line and generate the DB classes.
3) Run it and everything will be ok.
4) Uncomment line in step 3 and generate classes again.
5) Run it and you'll see that the exception will be at line "db.Update(row);" when linq2db will try to convert custom type Blob16AsGuidType to byte[].

If you change linq2db to prior 3.0.0 - to 2.9.8 for example then it will work wth and without GenerateDataTypes option.

### Environment details
linq2db version: 3.0.0+
Database Server: sqlite
Database Provider: sqlite
Operating system: windows 10 pro 21h1
.NET Framework: 4.8
[Linq2DbConvertionError.zip](https://github.com/linq2db/linq2db/files/6681953/Linq2DbConvertionError.zip)

**Comment by sdanyliv** [comment]
Thanks for reporting, will check.

**Comment by MaceWindu** [comment]
Was fixed in 3.4.1
